Cannot build from Git with 3.14

```
(venv) cc@pretty:~$ pip install git+https://github.com/KLayout/klayout
Collecting git+https://github.com/KLayout/klayout
  Cloning https://github.com/KLayout/klayout to /tmp/pip-req-build-hoeq1lhw
  Running command git clone --filter=blob:none --quiet https://github.com/KLayout/klayout /tmp/pip-req-build-hoeq1lhw
  Resolved https://github.com/KLayout/klayout to commit 56d91acc1523facf57e4bcc51af654bb24cfd828
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: klayout
  Building wheel for klayout (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for klayout (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [280 lines of output]
      /tmp/pip-build-env-pf8xwmvu/overlay/lib/python3.14t/site-packages/setuptools/dist.py:761: SetuptoolsDeprecationWarning: License classifiers are deprecated.
      !!
      
              ********************************************************************************
              Please consider removing the following classifiers in favor of a SPDX license expression:
      
              License :: OSI Approved :: GNU General Public License v3 (GPLv3)
      
              See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
              ********************************************************************************
      
      !!
        self._finalize_license_expression()
      running bdist_wheel
      running build
      running build_py
      creating build/lib.linux-x86_64-cpython-314t/pya
      copying src/pymod/distutils_src/pya/__init__.py -> build/lib.linux-x86_64-cpython-314t/pya
      creating build/lib.linux-x86_64-cpython-314t/klayout
      copying src/pymod/distutils_src/klayout/__init__.py -> build/lib.linux-x86_64-cpython-314t/klayout
      creating build/lib.linux-x86_64-cpython-314t/klayout/tl
      copying src/pymod/distutils_src/klayout/tl/__init__.py -> build/lib.linux-x86_64-cpython-314t/klayout/tl
      creating build/lib.linux-x86_64-cpython-314t/klayout/rdb
      copying src/pymod/distutils_src/klayout/rdb/__init__.py -> build/lib.linux-x86_64-cpython-314t/klayout/rdb
      creating build/lib.linux-x86_64-cpython-314t/klayout/pya
      copying src/pymod/distutils_src/klayout/pya/__init__.py -> build/lib.linux-x86_64-cpython-314t/klayout/pya
      creating build/lib.linux-x86_64-cpython-314t/klayout/lib
      copying src/pymod/distutils_src/klayout/lib/__init__.py -> build/lib.linux-x86_64-cpython-314t/klayout/lib
      creating build/lib.linux-x86_64-cpython-314t/klayout/lay
      copying src/pymod/distutils_src/klayout/lay/__init__.py -> build/lib.linux-x86_64-cpython-314t/klayout/lay
      creating build/lib.linux-x86_64-cpython-314t/klayout/db
      copying src/pymod/distutils_src/klayout/db/pcell_declaration_helper.py -> build/lib.linux-x86_64-cpython-314t/klayout/db
      copying src/pymod/distutils_src/klayout/db/__init__.py -> build/lib.linux-x86_64-cpython-314t/klayout/db
      running egg_info
      writing src/pymod/distutils_src/klayout.egg-info/PKG-INFO
      writing dependency_links to src/pymod/distutils_src/klayout.egg-info/dependency_links.txt
      writing top-level names to src/pymod/distutils_src/klayout.egg-info/top_level.txt
      reading manifest file 'src/pymod/distutils_src/klayout.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no files found matching '*.cc' under directory 'src/version'
      warning: no files found matching '*.cc' under directory 'src/plugins/*/db_plugin'
      warning: no files found matching '*.h' under directory 'src/plugins/*/db_plugin'
      warning: no files found matching 'src/py.typed'
      adding license file 'LICENSE'
      writing manifest file 'src/pymod/distutils_src/klayout.egg-info/SOURCES.txt'
      copying src/pymod/distutils_src/klayout/dbcore.pyi -> build/lib.linux-x86_64-cpython-314t/klayout
      copying src/pymod/distutils_src/klayout/laycore.pyi -> build/lib.linux-x86_64-cpython-314t/klayout
      copying src/pymod/distutils_src/klayout/libcore.pyi -> build/lib.linux-x86_64-cpython-314t/klayout
      copying src/pymod/distutils_src/klayout/py.typed -> build/lib.linux-x86_64-cpython-314t/klayout
      copying src/pymod/distutils_src/klayout/rdbcore.pyi -> build/lib.linux-x86_64-cpython-314t/klayout
      copying src/pymod/distutils_src/klayout/tlcore.pyi -> build/lib.linux-x86_64-cpython-314t/klayout
      running build_ext
      building 'klayout._tl' extension
      creating build/temp.linux-x86_64-cpython-314t/src/tl/tl
      Compiling with 8 threads.
      Building build/temp.linux-x86_64-cpython-314t/src/tl/tl/tlArch.o
      Building build/temp.linux-x86_64-cpython-314t/src/tl/tl/tlBase64.o
      Building build/temp.linux-x86_64-cpython-314t/src/tl/tl/tlColor.o
      Building build/temp.linux-x86_64-cpython-314t/src/tl/tl/tlCopyOnWrite.o
      Building build/temp.linux-x86_64-cpython-314t/src/tl/tl/tlDeferredExecution.o
      Building build/temp.linux-x86_64-cpython-314t/src/tl/tl/tlEnv.o
      Building build/temp.linux-x86_64-cpython-314t/src/tl/tl/tlEvents.oBuilding
       build/temp.linux-x86_64-cpython-314t/src/tl/tl/tlExceptions.o
...
      g++ -fno-strict-overflow -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -O3 -fPIC -DHAVE_CURL=1 -DHAVE_EXPAT=1 -DHAVE_PNG=1 -DKLAYOUT_MAJOR_VERSION=0 -DKLAYOUT_MINOR_VERSION=30 -DGSI_ALIAS_INSPECT=1 -DMAKE_PYA_LIBRARY=1 -DKLAYOUT_VERSION=0.30.1 -Isrc/version -Isrc/tl/tl -Isrc/gsi/gsi -I/home/cc/venv/include -I/usr/include/python3.14t -c src/pya/pya/pyaInternal.cc -o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaInternal.o -Wno-strict-aliasing -std=c++11
      src/pya/pya/pyaCallables.cc: In function ‘void pya::pya_object_deallocate(PyObject*)’:
      src/pya/pya/pyaCallables.cc:52:11: error: ‘PyObject’ {aka ‘struct _object’} has no member named ‘ob_refcnt’
         52 |   ++self->ob_refcnt;
            |           ^~~~~~~~~
      src/pya/pya/pya.cc: In constructor ‘pya::PythonInterpreter::PythonInterpreter(bool)’:
      src/pya/pya/pya.cc:330:21: warning: ‘void Py_SetProgramName(const wchar_t*)’ is deprecated [-Wdeprecated-declarations]
        330 |   Py_SetProgramName (const_cast<wchar_t *> (mp_py3_app_name.c_str ()));
            |   ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      In file included from /usr/include/python3.14t/Python.h:125,
                       from src/pya/pya/pya.cc:23:
      /usr/include/python3.14t/pylifecycle.h:37:38: note: declared here
         37 | Py_DEPRECATED(3.11) PyAPI_FUNC(void) Py_SetProgramName(const wchar_t *);
            |                                      ^~~~~~~~~~~~~~~~~
      src/pya/pya/pya.cc:337:19: warning: ‘void PySys_SetArgvEx(int, wchar_t**, int)’ is deprecated [-Wdeprecated-declarations]
        337 |   PySys_SetArgvEx (1, argv, 0);
            |   ~~~~~~~~~~~~~~~~^~~~~~~~~~~~
      In file included from /usr/include/python3.14t/Python.h:127:
      /usr/include/python3.14t/sysmodule.h:11:38: note: declared here
         11 | Py_DEPRECATED(3.11) PyAPI_FUNC(void) PySys_SetArgvEx(int, wchar_t **, int);
            |                                      ^~~~~~~~~~~~~~~
      Building build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaCallables.o has failed. Trying again.
      Building build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaCallables.o
      g++ -fno-strict-overflow -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -O3 -fPIC -DHAVE_CURL=1 -DHAVE_EXPAT=1 -DHAVE_PNG=1 -DKLAYOUT_MAJOR_VERSION=0 -DKLAYOUT_MINOR_VERSION=30 -DGSI_ALIAS_INSPECT=1 -DMAKE_PYA_LIBRARY=1 -DKLAYOUT_VERSION=0.30.1 -Isrc/version -Isrc/tl/tl -Isrc/gsi/gsi -I/home/cc/venv/include -I/usr/include/python3.14t -c src/pya/pya/pyaCallables.cc -o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaCallables.o -Wno-strict-aliasing -std=c++11
      Building build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaMarshal.o
      g++ -fno-strict-overflow -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -O3 -fPIC -DHAVE_CURL=1 -DHAVE_EXPAT=1 -DHAVE_PNG=1 -DKLAYOUT_MAJOR_VERSION=0 -DKLAYOUT_MINOR_VERSION=30 -DGSI_ALIAS_INSPECT=1 -DMAKE_PYA_LIBRARY=1 -DKLAYOUT_VERSION=0.30.1 -Isrc/version -Isrc/tl/tl -Isrc/gsi/gsi -I/home/cc/venv/include -I/usr/include/python3.14t -c src/pya/pya/pyaMarshal.cc -o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaMarshal.o -Wno-strict-aliasing -std=c++11
      Building build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaModule.o
      g++ -fno-strict-overflow -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -O3 -fPIC -DHAVE_CURL=1 -DHAVE_EXPAT=1 -DHAVE_PNG=1 -DKLAYOUT_MAJOR_VERSION=0 -DKLAYOUT_MINOR_VERSION=30 -DGSI_ALIAS_INSPECT=1 -DMAKE_PYA_LIBRARY=1 -DKLAYOUT_VERSION=0.30.1 -Isrc/version -Isrc/tl/tl -Isrc/gsi/gsi -I/home/cc/venv/include -I/usr/include/python3.14t -c src/pya/pya/pyaModule.cc -o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaModule.o -Wno-strict-aliasing -std=c++11
      Building build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaObject.o
      g++ -fno-strict-overflow -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -O3 -fPIC -DHAVE_CURL=1 -DHAVE_EXPAT=1 -DHAVE_PNG=1 -DKLAYOUT_MAJOR_VERSION=0 -DKLAYOUT_MINOR_VERSION=30 -DGSI_ALIAS_INSPECT=1 -DMAKE_PYA_LIBRARY=1 -DKLAYOUT_VERSION=0.30.1 -Isrc/version -Isrc/tl/tl -Isrc/gsi/gsi -I/home/cc/venv/include -I/usr/include/python3.14t -c src/pya/pya/pyaObject.cc -o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaObject.o -Wno-strict-aliasing -std=c++11
      src/pya/pya/pyaCallables.cc: In function ‘void pya::pya_object_deallocate(PyObject*)’:
      src/pya/pya/pyaCallables.cc:52:11: error: ‘PyObject’ {aka ‘struct _object’} has no member named ‘ob_refcnt’
         52 |   ++self->ob_refcnt;
            |           ^~~~~~~~~
      Building build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaRefs.o
      g++ -fno-strict-overflow -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -O3 -fPIC -DHAVE_CURL=1 -DHAVE_EXPAT=1 -DHAVE_PNG=1 -DKLAYOUT_MAJOR_VERSION=0 -DKLAYOUT_MINOR_VERSION=30 -DGSI_ALIAS_INSPECT=1 -DMAKE_PYA_LIBRARY=1 -DKLAYOUT_VERSION=0.30.1 -Isrc/version -Isrc/tl/tl -Isrc/gsi/gsi -I/home/cc/venv/include -I/usr/include/python3.14t -c src/pya/pya/pyaRefs.cc -o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaRefs.o -Wno-strict-aliasing -std=c++11
      Building build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaSignalHandler.o
      g++ -fno-strict-overflow -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -O3 -fPIC -DHAVE_CURL=1 -DHAVE_EXPAT=1 -DHAVE_PNG=1 -DKLAYOUT_MAJOR_VERSION=0 -DKLAYOUT_MINOR_VERSION=30 -DGSI_ALIAS_INSPECT=1 -DMAKE_PYA_LIBRARY=1 -DKLAYOUT_VERSION=0.30.1 -Isrc/version -Isrc/tl/tl -Isrc/gsi/gsi -I/home/cc/venv/include -I/usr/include/python3.14t -c src/pya/pya/pyaSignalHandler.cc -o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaSignalHandler.o -Wno-strict-aliasing -std=c++11
      Building build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaCallables.o has failed. Trying again.
      Building build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaStatusChangedListener.o
      g++ -fno-strict-overflow -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -O3 -fPIC -DHAVE_CURL=1 -DHAVE_EXPAT=1 -DHAVE_PNG=1 -DKLAYOUT_MAJOR_VERSION=0 -DKLAYOUT_MINOR_VERSION=30 -DGSI_ALIAS_INSPECT=1 -DMAKE_PYA_LIBRARY=1 -DKLAYOUT_VERSION=0.30.1 -Isrc/version -Isrc/tl/tl -Isrc/gsi/gsi -I/home/cc/venv/include -I/usr/include/python3.14t -c src/pya/pya/pyaStatusChangedListener.cc -o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaStatusChangedListener.o -Wno-strict-aliasing -std=c++11
      Building build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaUtils.o
      g++ -fno-strict-overflow -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -O3 -fPIC -DHAVE_CURL=1 -DHAVE_EXPAT=1 -DHAVE_PNG=1 -DKLAYOUT_MAJOR_VERSION=0 -DKLAYOUT_MINOR_VERSION=30 -DGSI_ALIAS_INSPECT=1 -DMAKE_PYA_LIBRARY=1 -DKLAYOUT_VERSION=0.30.1 -Isrc/version -Isrc/tl/tl -Isrc/gsi/gsi -I/home/cc/venv/include -I/usr/include/python3.14t -c src/pya/pya/pyaUtils.cc -o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaUtils.o -Wno-strict-aliasing -std=c++11
      src/pya/pya/pyaSignalHandler.cc: In member function ‘pya::PythonRef pya::CallbackFunction::callable() const’:
      src/pya/pya/pyaSignalHandler.cc:66:42: warning: ‘PyObject* PyWeakref_GetObject(PyObject*)’ is deprecated [-Wdeprecated-declarations]
         66 |     PyObject *self = PyWeakref_GetObject (m_weak_self.get ());
            |                      ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
      In file included from /usr/include/python3.14t/Python.h:115,
                       from src/pya/pya/pyaSignalHandler.h:27,
                       from src/pya/pya/pyaSignalHandler.cc:23:
      /usr/include/python3.14t/weakrefobject.h:30:44: note: declared here
         30 | Py_DEPRECATED(3.13) PyAPI_FUNC(PyObject *) PyWeakref_GetObject(PyObject *ref);
            |                                            ^~~~~~~~~~~~~~~~~~~
      src/pya/pya/pyaSignalHandler.cc: In member function ‘PyObject* pya::CallbackFunction::self_ref() const’:
      src/pya/pya/pyaSignalHandler.cc:90:30: warning: ‘PyObject* PyWeakref_GetObject(PyObject*)’ is deprecated [-Wdeprecated-declarations]
         90 |   return PyWeakref_GetObject (m_weak_self.get ());
            |          ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
      /usr/include/python3.14t/weakrefobject.h:30:44: note: declared here
         30 | Py_DEPRECATED(3.13) PyAPI_FUNC(PyObject *) PyWeakref_GetObject(PyObject *ref);
            |                                            ^~~~~~~~~~~~~~~~~~~
      g++ -fno-strict-overflow -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -O3 -shared build/temp.linux-x86_64-cpython-314t/src/pya/pya/gsiDeclPya.o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pya.o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaCallables.o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaConvert.o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaHelpers.o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaInspector.o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaInternal.o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaMarshal.o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaModule.o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaObject.o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaRefs.o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaSignalHandler.o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaStatusChangedListener.o build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaUtils.o build/lib.linux-x86_64-cpython-314t/klayout/lib_tl.cpython-314t-x86_64-linux-gnu.so build/lib.linux-x86_64-cpython-314t/klayout/lib_gsi.cpython-314t-x86_64-linux-gnu.so -L/usr/lib64 -o build/lib.linux-x86_64-cpython-314t/klayout/lib_pya.cpython-314t-x86_64-linux-gnu.so -Wl,-soname,lib_pya.cpython-314t-x86_64-linux-gnu.so -Wl,-rpath,$ORIGIN -Wl,--strip-all
      /usr/bin/ld: cannot find build/temp.linux-x86_64-cpython-314t/src/pya/pya/pyaCallables.o: No such file or directory
      collect2: error: ld returned 1 exit status
      error: command '/opt/gcc-latest/bin/g++' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for klayout
Failed to build klayout
ERROR: Failed to build installable wheels for some pyproject.toml based projects (klayout)
```

--- a/src/pya/pya/pyaCallables.cc
+++ b/src/pya/pya/pyaCallables.cc
@@ -49,7 +49,7 @@ pya_object_deallocate (PyObject *self)
   //  may trigger a GC (https://github.com/KLayout/klayout/issues/1054).
   //  According to the comments this may be turned into a release mode assertion, so
   //  we better work around it.
-  ++self->ob_refcnt;
+  Py_IncRef(self);
 
   //  Mute Python warnings in debug case
   PyObject_GC_UnTrack (self);

